eo-maven-plugin/src/main/java/org/eolang/maven/ChNarrow.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+/**
+ * Short version of hash.
+ *
+ * @since 0.28.11
+ */
+final class ChNarrow implements CommitHash {
+
+    /**
+     * Delegate.
+     */
+    private final CommitHash full;
+
+    /**
+     * The main constructor.
+     *
+     * @param full Delegate
+     */
+    ChNarrow(final CommitHash full) {
```

Comment:
@volodya-lombrozo what if `ChNarrow` hash is passed as `full`?

Reply:
@mximp Done.

Reply:
@volodya-lombrozo I still think we have a design issue here. Constructor declares no restriction on `CommitHash` and as a client of this class I assume any implementation can be passed. Whereas logic implies particular implementation.
Since the whole class is just a wrapper for `String.substring()` method, don't you think typing this several times would be easier and neater?

Reply:
@mximp No, I don't think so. Please, see `validHash()` function inside the `ChNarrow` class. It's quite a lot of logic here. We didn't validate any hashes before, it means our code becomes even better with that PR.